niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/config/RuntimeDependentPermissionConfiguration.java

```diff
@@ -43,20 +76,57 @@
  */
 class RuntimeDependentPermissionConfiguration implements TestConfiguration {
     private static final String TAG = "PermissionTester";
-    private static Logger sLogger = LoggerFactory.createDefaultLogger(TAG);
+    private static final Logger sLogger = LoggerFactory.createDefaultLogger(TAG);
 
-    private Activity mActivity;
+    private final Activity mActivity;
     private CountDownLatch mCountDownLatch;
 
-    private static final String[] REQUIRED_PERMISSIONS = new String[]{
-            Manifest.permission.READ_EXTERNAL_STORAGE,
-            Manifest.permission.READ_PHONE_NUMBERS,
+    private static final String[] RUNTIME_SDK33 = new String[]{
+            NEARBY_WIFI_DEVICES,
+            POST_NOTIFICATIONS,
+            READ_MEDIA_AUDIO,
+            READ_MEDIA_IMAGES,
+            READ_MEDIA_VIDEO
     };
-
-    private static final String[] PERMISSIONS_UNDER_TEST = new String[]{
-            Manifest.permission.ACCESS_MEDIA_LOCATION,
-            Manifest.permission.SEND_SMS,
+    private static final String[] RUNTIME_SDK31 = new String[]{
+            BLUETOOTH_ADVERTISE,
+            BLUETOOTH_CONNECT,
+            BLUETOOTH_SCAN,
+            UWB_RANGING
+    };
+    private static final String[] RUNTIME_SDK29 = new String[]{
+            ACCESS_MEDIA_LOCATION,
+            ACTIVITY_RECOGNITION
+    };
+    private static final String[] RUNTIME_SDK28A = new String[]{
+            ACCESS_COARSE_LOCATION,
+            ACCESS_FINE_LOCATION,
+            ANSWER_PHONE_CALLS,
+            BODY_SENSORS,
+            CALL_PHONE,
+            CAMERA,
     };
+    private static final String[] RUNTIME_SDK28B = new String[]{
+            READ_CALENDAR,
+            READ_CALL_LOG,
+            READ_CONTACTS,
+            //READ_EXTERNAL_STORAGE,//The permission does not show user consent dialogue as of 33
+            READ_PHONE_NUMBERS,
+            READ_PHONE_STATE,
+            READ_SMS,
+    };
+    private static final String[] RUNTIME_SDK28C = new String[]{
+            WRITE_CALENDAR,
+            WRITE_CALL_LOG,
+            WRITE_CONTACTS,
+            //WRITE_EXTERNAL_STORAGE,//The permission does not show user consent dialogue
+            ADD_VOICEMAIL
+    };
+    private static final String[] REQUIRED_PERMISSIONS = RUNTIME_SDK29;// new String[]{
```

Comment:
With the change to this and `PERMISSIONS_UNDER_TEST` below, do these tests still work on older SDK versions? The original intent was to grant only the required permissions to test the cases where the permissions under test were not granted, but with these changes it looks like these two variables have the same permissions assigned to them.

In general, how are the other RUNTIME_SDKxx constants used? I didn't see them referenced anywhere else.

Reply:
Thanks! Are all the static permission imports still in use, or can those be removed as well?

Reply:
removed.

Reply:
Looks good; thanks!